File: drift/src/main/java/drift/com/drift/helpers/Alert.java
package drift.com.drift.helpers;

import android.app.Activity;
import android.content.DialogInterface;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;


public class Alert {

    public Alert() {}

    public static void showAlert(Activity activity, String title, String message, String positiveTitle, @Nullable final Runnable block) {
        if (activity == null || activity.isFinishing()) {
            return;
        }
        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
        builder.setMessage(message);
        builder.setTitle(title)
                .setPositiveButton(positiveTitle, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        if (block != null) {
                            block.run();
                        }
                    }
                })
                .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                    }
                }).create().show();
    }

}

File: drift/src/main/java/drift/com/drift/helpers/Alert.kt
package drift.com.drift.helpers

import android.app.Activity
import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog


class Alert {
    companion object {

        fun showAlert(activity: Activity?, title: String, message: String, positiveTitle: String, block: (() -> Unit)?) {
            if (activity == null || activity.isFinishing) {
                return
            }
            val builder = AlertDialog.Builder(activity)
            builder.setMessage(message)
            builder.setTitle(title)
                    .setPositiveButton(positiveTitle) { _, _ ->
                        block?.invoke()
                    }
                    .setNegativeButton("Cancel") { _, _ -> }.create().show()
        }
    }

}